### ISSUE TITLE
Encrypted wallet shows stealth addresses without unlocking

### ISSUE THREAD

**Issue by radeksalomon** [vergecurrency/VERGE#451]
When I moved my wallet from OS X to Windows 10 moving the backup .dat file, I was able to open the file in the wallet app and then I was able to see all addresses including stealth addresses before I unlocked the wallet - it was encrypted on OS X. Seems it is the issue with design of the wallet app, but hope there is some solution. Now when somebody obtain my .dat file, he is able to see the stealth addresses even if the wallet is encrypted.

**Comment by justinvforvendetta** [comment]
osx didnt show wallet addresses when it was encrypted? but windows does?

**Comment by radeksalomon** [comment]
Good point. Also OS X Wallet shows the addresses when the wallet is locked. It seems the app has bad design for this purpose. Is it able to somehow fix it? Now if someone gets my .dat file, he will know all my addresses even the stealth addresses.

**Comment by justinvforvendetta** [comment]
well it is not broken. no one would be able to access it, but we can work on a way to hide the addresses if it Is encrypted.

**Comment by Nissemus** [comment]
If you're going to hide the stealth addresses it would be a good idea to also hide the amounts associated with those addresses.

**Comment by justinvforvendetta** [comment]
the stealth addresses aren't static, if you were under that impression

**Comment by Nissemus** [comment]
My point is that if a bad actor is looking for proof of a particular stealth transaction they may be able to identify it by the amount, if the amounts are visible in an unencrypted wallet.
